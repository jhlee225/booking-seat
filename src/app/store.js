import { configureStore } from "@reduxjs/toolkit";
import loginReducer from "../features/login/loginSlice";
import seatReducer from "../features/seat/seatSlice";
import menuReducer from "../features/menu/menuSlice";
import bookReducer from "../features/booking/BookSlice";
import timeReducer from "../features/time/timeSlice";
import signUpReducer from "../features/signup/signupSlice";

export default configureStore({
  reducer: {
    login: loginReducer,
    seat: seatReducer,
    menu: menuReducer,
    time: timeReducer,
    book: bookReducer,
    signUp: signUpReducer,
  },
});
