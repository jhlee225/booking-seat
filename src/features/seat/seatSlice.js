import { createSlice } from "@reduxjs/toolkit";

export const seatSlice = createSlice({
  name: "seat",
  initialState: {
    floor: { floor: "1층 좌측", row: 5, col: 5 },
    seats: [
      [
        "blank",
        { seatNum: 1, isIn: false },
        { seatNum: 2, isIn: false },
        { seatNum: 3, isIn: false },
        { seatNum: 4, isIn: false },
      ],
      [{ seatNum: 5, isIn: false }, "blank", "blank", "blank", "blank"],
      [
        { seatNum: 6, isIn: false },
        "blank",
        { seatNum: 7, isIn: false },
        { seatNum: 8, isIn: false },
        { seatNum: 9, isIn: false },
      ],
    ],
  },

  reducers: {
    setFloor: (state, action) => {
      state.floor = action.payload.floor;
    },
    setSeats: (state, action) => {
      state.seats = action.payload.seats;
    },
  },
});

export const { setFloor, setSeats } = seatSlice.actions;

export const selectFloor = (state) => state.seat.floor;
export const selectSeats = (state) => state.seat.seats;

export default seatSlice.reducer;
