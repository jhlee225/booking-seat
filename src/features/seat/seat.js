import React from "react";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";
import { selectIsLogin } from "../menu/menuSlice";
import { selectFloor, selectSeats } from "./seatSlice";
export function Seats() {
  const floor = useSelector(selectFloor);
  const seats = useSelector(selectSeats);
  return (
    <>
      <div className="seats">
        <header className="floorName">{floor.floor}</header>
        <main className="seatTable">
          {seats.map((row, key) => (
            <SeatRow data={row} key={key} />
          ))}
        </main>
        <Link to="/">
          <div className="back btn">메인메뉴</div>
        </Link>
      </div>
    </>
  );
}

function SeatRow(props) {
  const { data } = props;
  return (
    <div className="row">
      {data.map((seat, key) =>
        seat === "blank" ? (
          <div key={key} className="blank"></div>
        ) : (
          <Seat key={key} data={seat} />
        )
      )}
    </div>
  );
}

function Seat(props) {
  const { data } = props;
  const isLogin = useSelector(selectIsLogin);
  return (
    <>
      <Link to={isLogin ? "/Time" : "/Login"}>
        {data.isIn ? (
          <div className="seat userIn">{data.seatNum}</div>
        ) : (
          <div className="seat">{data.seatNum}</div>
        )}
      </Link>
    </>
  );
}
