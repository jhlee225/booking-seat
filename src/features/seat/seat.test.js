import React from "react";
import { render } from "@testing-library/react";
import { HashRouter as Router, Route } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../app/store";
import { Seats } from "./seat";

it("render Seats", () => {
  const seatsRendered = render(
    <Provider store={store}>
      <Router>
        <Seats />
      </Router>
    </Provider>
  );
  expect(seatsRendered.container).toMatchSnapshot();
});
