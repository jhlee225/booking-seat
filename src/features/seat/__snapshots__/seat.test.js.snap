// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Seats 1`] = `
<div>
  <div
    class="seats"
  >
    <header
      class="floorName"
    >
      1층 좌측
    </header>
    <main
      class="seatTable"
    >
      <div
        class="row"
      >
        <div
          class="blank"
        />
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            1
          </div>
        </a>
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            2
          </div>
        </a>
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            3
          </div>
        </a>
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            4
          </div>
        </a>
      </div>
      <div
        class="row"
      >
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            5
          </div>
        </a>
        <div
          class="blank"
        />
        <div
          class="blank"
        />
        <div
          class="blank"
        />
        <div
          class="blank"
        />
      </div>
      <div
        class="row"
      >
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            6
          </div>
        </a>
        <div
          class="blank"
        />
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            7
          </div>
        </a>
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            8
          </div>
        </a>
        <a
          href="#/Login"
        >
          <div
            class="seat"
          >
            9
          </div>
        </a>
      </div>
    </main>
    <a
      href="#/"
    >
      <div
        class="back btn"
      >
        메인메뉴
      </div>
    </a>
  </div>
</div>
`;
